{
    "identifier": "CVE-2019-10099",
    "summary": "Prior to Spark 2.3.3, in certain situations Spark would write user data to local disk unencrypted, even if spark.io.encryption.enabled=true. This includes cached blocks that are fetched to disk (controlled by spark.maxRemoteBlockSizeFetchToMem); in SparkR, using parallelize; in Pyspark, using broadcast and parallelize; and use of python udfs. See: CVE-2019-10099.",
    "references": [
        {
            "url": "",
            "reference_id": "pyup.io-37352"
        }
    ],
    "vulnerable_packages": [
        "pkg:pypi/pyspark@2.1.3",
        "pkg:pypi/pyspark@2.3.0",
        "pkg:pypi/pyspark@2.2.3",
        "pkg:pypi/pyspark@2.1.1",
        "pkg:pypi/pyspark@2.1.2",
        "pkg:pypi/pyspark@2.2.0",
        "pkg:pypi/pyspark@2.3.1",
        "pkg:pypi/pyspark@2.2.1",
        "pkg:pypi/pyspark@2.3.2",
        "pkg:pypi/pyspark@2.2.2"
    ],
    "resolved_packages": [
        "pkg:pypi/pyspark@2.3.4",
        "pkg:pypi/pyspark@3.0.1",
        "pkg:pypi/pyspark@2.4.5",
        "pkg:pypi/pyspark@2.4.3",
        "pkg:pypi/pyspark@2.4.7",
        "pkg:pypi/pyspark@3.0.0",
        "pkg:pypi/pyspark@2.4.0",
        "pkg:pypi/pyspark@2.3.3",
        "pkg:pypi/pyspark@2.4.1",
        "pkg:pypi/pyspark@2.4.4",
        "pkg:pypi/pyspark@2.4.2",
        "pkg:pypi/pyspark@2.4.6"
    ]
}